### PR TITLE
[gui] [vulkan] Modify some internal APIs

### PR DESCRIPTION
**Pull request by Morcki**
ChangeList:
1. Remove old API function `dump_depth_buffer` from swap_chain.h. Still keep `Window.get_depth_buffer_as_numpy` (usage is same with #5410 ). And there is also another API function `get_depth_buffer` which support multi-backends (details refer to #5455 ).
2. Fix a bug about `Scene.mesh` when there is no indices, it would get a bug.

### CODE REVIEW

**Review by YuCrazing — APPROVED** [review]
LGTM!

File: python/taichi/ui/scene.py [hunk 1]
```diff
@@ -142,7 +142,10 @@ def mesh(self,
         if vertex_count is None:
             vertex_count = vertices.shape[0]
         if index_count is None:
-            index_count = indices.shape[0]
+            if indices is None:
+                index_count = vertex_count  # FIXME : Need to confirm
+            else:
+                index_count = indices.shape[0]
```
Comment (PENGUINLIONG) [inline python/taichi/ui/scene.py:148]: probably the following?
```python
if indices is not None:
    if index_count is None:
        index_count = indices.shape[0]
```

**Review by PENGUINLIONG — APPROVED** [review]
lgtm despite a minor tweak

File: python/taichi/ui/scene.py (hunk 1 above)
Comment (Morcki) [reply python/taichi/ui/scene.py:148]: I think we must pass a value to index_count.

Comment (Morcki) [reply python/taichi/ui/scene.py:148]: I tried to pass this value using None, but failed to get an error information. It's the reason that pybind donnot support to passing None value to int object.

**Review by ailzhang — APPROVED** [review]
LGTM！

**Pull request merged**